run_axelrod

```diff
@@ -1,135 +1,147 @@
-        help='exclude cheating strategies plot')
-
-    parser.add_argument(
-        '--xa', "--exclude-combined",
-        action='store_true',
-        dest="exclude_combined",
-        help='exclude combined strategies plot')
-
-    parser.add_argument(
-        '--ne', "--no-ecological",
-        action='store_true',
-        dest="no_ecological",
-        help='no ecological variant')
-
-    parser.add_argument(
-        '-p', '--processes',
-        type=int,
-        default=None,
-        help='Number of parallel processes to spawn. 0 uses cpu count.')
-
-    parser.add_argument(
-        '--rc', "--rebuild-cache",
-        action='store_true',
-        dest="rebuild_cache",
-        help='rebuild cache and save to file')
-
-    parser.add_argument(
-        '-c', '--cache_file',
-        type=str,
-        default='./cache.txt',
-        help='Path to cache file')
-
-    parser.add_argument(
-        '-n', '--noise',
-        type=float,
-        default=0,
-        help='Noise level')
-
-    parser.add_argument(
-        '-i', '--image_format',
-        type=str,
-        default="svg",
-        help='Image format for matplotlib, e.g. svg, jpeg, png')
-
-    args = parser.parse_args()
-
-    if all([args.exclude_basic,
-            args.exclude_ordinary,
-            args.exclude_cheating,
-            args.exclude_combined]):
-        print("You've excluded everything - nothing for me to do")
     else:
-        setup_logging(args.logging_destination, args.verbosity)
-        # Unravel argparse Namespace object to python keyword arguments.
-        kwargs = vars(args)
-        del kwargs["logging_destination"]
-        del kwargs["verbosity"]
-
-        if kwargs['prob_end'] is None:
-            del kwargs["prob_end"]
-            run_tournaments(**kwargs)
-        else:
-            del kwargs["turns"]
-            run_prob_end_tournaments(**kwargs)
+        tournament = axl.ProbEndTournament(players, repetitions=repetitions,
+                                           turns=turns, noise=noise,
+                                           processes=processes,
+                                           deterministic_cache=cache)
+        label = "Turns: {}, repetitions: {}, noise: {}, strategies: {}. ".format(turns, tournament.repetitions, noise, len(tournament.nplayers))
+
+    return tournament.play(), label
+
+
+if __name__ == '__main__':
+    arguments = docopt(__doc__, version='Run Axelrod 0.1')
+
+    # Parse arguments
+    turns = int(arguments['-t'])
+    repetitions = int(arguments['-r'])
+    prob_end = float(arguments['-e'])
+    noise = float(arguments['-n'])
+    processes = eval(arguments['-p'])
+
+    output_directory = arguments['-o']
+    if output_directory == "None":
+        output_directory = None
+    cache_file = arguments['-c']
+    if cache_file == "None":
+        cache_file = None
+
+    image_format = arguments['-i']
+    rebuild_cache = arguments['--rc']
+    exclude_basic = arguments['--xb']
```

Comment:
Yup, I'm happy with do nothing by default myself (otherwise it's creating
files you might not want) and then just add in the options for what you
want.

I think the exclude made sense back in the day when this was a quick one
stop 'update everything' tool. Now that we have so many strategies and
counting perhaps it's not as apt.

On Wed, 20 Apr 2016, 07:22 Marc Harper, <notifications@github.com> wrote:

> In run_axelrod
> <https://github.com/Axelrod-Python/Axelrod/pull/557#discussion_r60355088>:
>
> > +    turns = int(arguments['-t'])
> > +    repetitions = int(arguments['-r'])
> > +    prob_end = float(arguments['-e'])
> > +    noise = float(arguments['-n'])
> > +    processes = eval(arguments['-p'])
> > +
> > +    output_directory = arguments['-o']
> > +    if output_directory == "None":
> > +        output_directory = None
> > +    cache_file = arguments['-c']
> > +    if cache_file == "None":
> > +        cache_file = None
> > +
> > +    image_format = arguments['-i']
> > +    rebuild_cache = arguments['--rc']
> > +    exclude_basic = arguments['--xb']
>
> I guess it depends on the purpose. Maybe running the basic tournament is
> fine, and it's quick anyway. But the other tournaments are much longer.
>
> It's just easier to understand if you have to tell it what you want it to
> do. Otherwise you have to know everything it wants to do by default and
> then scale it back.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/557/files/076d86d946b28d212aa4dfbc62ee563512bff5ba#r60355088>
>